testbed/addons/karpenter/construct.ts

```diff
@@ -0,0 +1,61 @@
+import cdk = require('@aws-cdk/core');
+import eks = require('@aws-cdk/aws-eks');
+import iam = require('@aws-cdk/aws-iam');
+
+export interface KarpenterProps {
+    cluster: eks.Cluster;
+    namespace: string;
+}
+
+export class KarpenterController extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string, props: KarpenterProps) {
+        super(scope, id);
+
+        const sa = props.cluster.addServiceAccount('karpenter-controller-sa', {
+            name: "karpenter",
+            namespace: props.namespace
+        });
+
+        sa.role.attachInlinePolicy(new iam.Policy(this, 'karpenter-controller-policy', {
+            statements: [
+                new iam.PolicyStatement({
+                    resources: ['*'],
+                    actions: ["ec2:CreateLaunchTemplate", "ec2:CreateFleet", "ec2:RunInstances", "ec2:CreateTags", "iam:PassRole", "ec2:TerminateInstances", "ec2:DescribeLaunchTemplates", "ec2:DescribeInstances", "ec2:DescribeSecurityGroups", "ec2:DescribeSubnets", "ec2:DescribeInstanceTypes", "ec2:DescribeInstanceTypeOfferings", "ec2:DescribeAvailabilityZones", "ssm:GetParameter"],
+                }),
+            ],
+        }));
+
+        props.cluster.awsAuth.addRoleMapping(new iam.Role(this, 'karpenter-node-role', {
+            assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+            managedPolicies: [
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')
+            ]
+        }),
+            {
+                username: `system:node:{{EC2PrivateDNSName}}`,
+                groups: ['system:bootstrappers', 'system:nodes']
+            }
+        )
+
+        const chart = props.cluster.addHelmChart('KarpenterHelmChart', {
+            chart: 'karpenter-v0.2.8.tgz',
```

Comment:
Are you sure you need to put .tgz in here? Can we just call this "karpenter" and the tgz will be looked up by the version?